### DISCUSSION TITLE
Potential bug

### DISCUSSION THREAD

**Discussion by andped10** [easyscience/reflectometry-lib#70] [General]
https://github.com/easyScience/EasyReflectometryLib/blob/71fbb174ea1cf1396172aaaa51f9feb345af27a1/EasyReflectometry/sample/items/surfactant_layer.py#L152

It seems wrong that the we are only manipulating layer 0.  

I would assume that it should have been layer 1 in the assignment. 

@arm61 are you able to clarify

**Comment by arm61** [comment]
This setter aims to make it so that the two roughnesses are the same. Can you test, using the https://github.com/easyScience/EasyReflectometryLib/blob/main/docs/monolayer.ipynb notebook, what happens if the roughness is allowed to vary and this is set to `True` and `False` and see if it works first? (It has been a while since I thought about this part of the code).

**Reply by andped10** [reply]
I will look into it

**Comment by andped10** [comment]
The code is correct.

I just had to understand how a constraint between an dependent and independent variable is set and how it functions.

When set the constraint "pass" the value when the independent variable is changed whereas it is ignored when the value is set for the dependent variable.
